### DISCUSSION TITLE
cygport/APKBUILD support

### DISCUSSION THREAD

**Discussion by Freed-Wu** [termux/termux-language-server#25] [Ideas]
https://cygwin.github.io/cygport/cygport_in.html
cygport use a similar syntax to gentoo ebuild:
cygport -> ebuild
cygclass -> eclass

Another case is APKBUILD, its syntax looks like PKGBUILD.

https://wiki.alpinelinux.org/wiki/APKBUILD_Reference

@qaqland any idea about APKBUILD?

**Comment by qaqland** [comment]
Compared to PKGBUILD, APKBUILD is a pure posix shell script. Its working directory is usually under ~/aports, which is cloned from https://gitlab.alpinelinux.org/alpine/aports.

I'm not sure about the difficulty. The existing useful tools are https://gitlab.alpinelinux.org/alpine/infra/atools-go and https://gitlab.alpinelinux.org/alpine/lua-aports

**Reply by Freed-Wu** [reply]
<img width="2470" height="1751" alt="Screenshot_20260919_130421" src="https://github.com/user-attachments/assets/a8f38a0a-c039-4653-90f1-2b26f9e9dd49" />

How to complete package name and hint package version?

**Reply by qaqland** [reply]
FYI https://gitlab.alpinelinux.org/alpine/apk-tools/-/tree/master/src/bash, most time there is no need to specify package version in depends.

**Reply by Freed-Wu** [reply]
To support APKBUILD, we need a specification to provide the variables/function names/document, and a python package to get package name/version, which is pyalpm or portage for PKGBUILD and ebuild.

**Reply by qaqland** [reply]
Here is the manual https://gitlab.alpinelinux.org/alpine/abuild/-/blob/master/APKBUILD.5.scd

Python 3 bindings for libapk is incomplete, only version comparison:

- https://gitlab.alpinelinux.org/alpine/apk-tools/-/work_items/11062
- https://github.com/alpinelinux/apk-tools/blob/master/python/apk_module.c

**Reply by Freed-Wu** [reply]
<img width="1019" height="211" alt="Screenshot_20260920_163444" src="https://github.com/user-attachments/assets/8736c4fc-e7ea-4ec6-bb43-08b257874a74" />

OK.

**Reply by Freed-Wu** [reply]
BTW, https://gitlab.alpinelinux.org/alpine/abuild/-/blob/master/APKBUILD.5.scd#L248-250 independant should be independent.
